app/javascript/components/applications/application_form/preferences/PreferencesSection.js

```diff
@@ -33,41 +34,44 @@ class PreferencesSection extends React.Component {
           autofillPreferences.push(editPreference)
         }
       })
-      formApi.setValue('preferences', autofillPreferences)
+      form.change('preferences', autofillPreferences)
     }
   }
 
   render () {
-    const { formApi, listingPreferences } = this.props
-    const fullHousehold = getFullHousehold(formApi.values)
-    const preferences = formApi.values.preferences
+    const { form, listingPreferences } = this.props
+    const fullHousehold = getFullHousehold(form.getState().values)
 
     return (
       <div className='border-bottom margin-bottom--2x'>
         <h3>Preferences</h3>
-        {
-          !isEmpty(preferences) && preferences.map((pref, i) => (
-            <div className='border-bottom margin-bottom--2x' key={i}>
-              <PreferenceForm {...{i, pref, formApi, listingPreferences, fullHousehold}} />
-            </div>
-          ))
-        }
-
-        <div className='row'>
-          <div className='form-group'>
-            <div className='small-4 columns'>
-              <button
-                onClick={() => formApi.addValue('preferences', '')}
-                disabled={disableAddPreference(formApi, listingPreferences)}
-                type='button'
-                className='mb-4 mr-4 btn btn-success'
-                id='add-preference-button'>
-                  + Add Preference
-              </button>
-            </div>
-          </div>
-        </div>
-
+        <FieldArray name='preferences'>
+          {({ fields }) =>
+            <React.Fragment>
+              { fields.map((name, index) => {
+                return (
+                  <div className='border-bottom margin-bottom--2x' key={name}>
+                    <PreferenceForm {...{index, name, form, listingPreferences, fullHousehold}} />
+                  </div>)
+              })
+              }
+              <div className='row'>
```

Comment:
@software-project does the row with the Add Preference button need to be included within the `FieldArray` component?

Reply:
@callaghanc we paired on this one with Chris. We had had some hard times to make it work and moving it here worked. Maybe there is some other way, to keep it outside, but we couldn't find it.